lambdas/process-mobile-app-callback/src/main/java/uk/gov/di/ipv/core/processmobileappcallback/ProcessMobileAppCallbackHandler.java

```diff
@@ -152,13 +152,13 @@ private void validateCallbackRequest(MobileAppCallbackRequest callbackRequest)
         }
     }
 
-    private void validateCriResponse(Optional<CriResponseItem> criResponse)
+    private void validateCriResponse(CriResponseItem criResponse, String state)
             throws InvalidMobileAppCallbackRequestException, InvalidCriResponseException {
-        if (criResponse.isEmpty()) {
+        if (criResponse == null || !Objects.equals(criResponse.getOauthState(), state)) {
```

Comment:
This feels more thorough as is using the criId and state to specify. Didn't need to change but feels slightly better